divi/src/masternode.h

```diff
@@ -274,6 +290,15 @@ class CMasternodeBroadcast : public CMasternode
         READWRITE(vin);
         READWRITE(addr);
         READWRITE(pubKeyCollateralAddress);
+        if (nVersion >= MN_REWARD_SCRIPT_VERSION) {
+            READWRITE(rewardScript);
+        } else {
+            if (ser_action.ForRead()) {
+                rewardScript = GetDefaultRewardScript();
+            } else if (rewardScript != GetDefaultRewardScript()) {
+                LogPrintf("WARNING: CMasternodeBroadcast - ignoring changed reward script for serialisation of %s\n", vin.prevout.ToString());
+            }
+        }
```

Comment:
The new reward script field should be placed at the end of the variable definitions, as well as its serialization to avoid becoming a hard fork

Reply:
As discussed, I think this is fine.  But I will work on explicit regtests to ensure compatibility, and mark this as draft for now.